src/zope/testing/doctestcase.py

```diff
@@ -0,0 +1,138 @@
+r"""Doctests in TestCase classes
+
+The original doctest unittest integration was based on unittest test
+suites, which have fallen out of favor. This module provides a way to
+define doctests inside of unittest TestCase classes. It also provides
+better integration with unittest test fixtures, because doctests use
+setup provided by the containing test case class.
+
+You can define doctests in 4 ways:
+
+- references to named files
+
+- strings
+
+- decorated functions with docstrings
+
+- reference to named files decorating test-specific setup functions
+
+Here are some examples::
+
+    import unittest
+    from zope.testing.doctestcase import DocTest
+
+    class MyTest(unittest.TestCase):
+
+        test1 = DocTest('test1.txt')
+
+        test2 = DocTest('''
+        >>> 1 + 1
+        2
+        ''')
+
+        @DocTest
+        def test3(self):
+            '''
+            >>> test.x
+            3
+            '''
+            self.x = 3
+
+        @DocTest('test4.txt')
+        def test4(self):
+            self.y = 5
+
+The exampe illustrates some additional points:
+
+- the test case ``self`` argument is exposed to the doctest as the
+  ``test`` global variable.  This gives the test access to any
+  attributes defined in the test case.
+
+- The body of a function who's docstring defines a doctest is run before
+  the test, providing test-specific setup, if desired.
+
+- A the return value of the doctest function can be used to decorate a
+  function.  The decorated function is run before the test is executed.
+  This provides a way to provide some test-specfic setup, if desired.
+
+  Note that the docstring, if any, of the decorated function is ignored.
```

Comment:
The docstring is obviously not ignored in the "naked decorator" case. So this sentence, in isolation, is slightly confusing.